tests/torch/test_training.py

```diff
@@ -0,0 +1,127 @@
+"""Tests for FHE training."""
```

Comment:
again here, are we sure we want to add training feature/tests in this PR ? I think I would have preferred it to be separated

Reply:
We need a test that makes sure we keep that test working all along training in FHE integration.